File: Python/自己搞定/Easy/226. Invert Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if root == None:
            return 
        # 將左右節點互換
        root.left,root.right = root.right,root.left
        # 去Son Node 持續做
        self.invertTree(root.left)
        self.invertTree(root.right)
        return root 

File: C++/自己搞定/Easy/226. Invert Binary Tree.cpp
#include<basetsd.h>
#include<iostream>
#include<deque>

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
    };
class Solution {
public:
    TreeNode* invertTree(TreeNode* root) {
        if(!root)return root;
        std::deque<TreeNode*> treeDeque{root};
        TreeNode* temp = nullptr;
        TreeNode* curr = nullptr;
        while(!treeDeque.empty()){
            curr = treeDeque.front();
            treeDeque.pop_front();
            if(curr->left)treeDeque.emplace_back(curr->left);
            if(curr->right)treeDeque.emplace_back(curr->right);
            temp = curr->left;
            curr->left =curr->right;
            curr->right = temp;      
        }
        return root;
    }
};